### PR TITLE
[exporter/bmchelix] Enrich metric names with datapoint attributes

### PR DESCRIPTION
**Pull request by NassimBtk**
#### Description
This pull request enhances the BMC Helix exporter to ensure metric names are uniquely identifiable in BMC Helix Operations Management by enriching them with datapoint attributes and applying normalization for compatibility. It also improves label handling, removes legacy enrichment logic, and updates tests to validate the new enrichment approach and normalization rules.

**Metric enrichment and normalization:**

* Metric names are now enriched with non-core datapoint attributes, appended as dot-separated suffixes, and normalized to match BMC Helix compatibility requirements. This ensures unique identification and prevents key collisions. (`exporter/bmchelixexporter/internal/operationsmanagement/metrics_producer.go`, [[1]](diffhunk://#diff-5c9d35d9692f79277d9122bf02f6425f12f066b0f1304356242b0715bfdacf9fR162-R180) [[2]](diffhunk://#diff-5c9d35d9692f79277d9122bf02f6425f12f066b0f1304356242b0715bfdacf9fR193-L185) [[3]](diffhunk://#diff-5c9d35d9692f79277d9122bf02f6425f12f066b0f1304356242b0715bfdacf9fL336-R409)
* Entity label values (`entityTypeId`, `entityName`) and metric names are normalized to remove invalid characters (e.g., colons) and conform to required patterns. Label values are also normalized (e.g., commas replaced with whitespace). (`exporter/bmchelixexporter/internal/operationsmanagement/metrics_producer.go`, [[1]](diffhunk://#diff-5c9d35d9692f79277d9122bf02f6425f12f066b0f1304356242b0715bfdacf9fL223-R250) [[2]](diffhunk://#diff-5c9d35d9692f79277d9122bf02f6425f12f066b0f1304356242b0715bfdacf9fL271-R303) [[3]](diffhunk://#diff-5c9d35d9692f79277d9122bf02f6425f12f066b0f1304356242b0715bfdacf9fL295-R331)

**Codebase simplification and removal of legacy logic:**

* The legacy `enrichMetricNamesWithAttributes` function is removed, replaced by the new `createEnrichedMetricWithDpAttributes` approach for metric enrichment. (`exporter/bmchelixexporter/internal/operationsmanagement/metrics_producer.go`, [[1]](diffhunk://#diff-5c9d35d9692f79277d9122bf02f6425f12f066b0f1304356242b0715bfdacf9fR193-L185) [[2]](diffhunk://#diff-5c9d35d9692f79277d9122bf02f6425f12f066b0f1304356242b0715bfdacf9fL336-R409)

#### Link to tracking issue
Fixes #46558

#### Testing
* Unit tests are updated and expanded to cover the new enrichment logic, normalization rules, and edge cases (e.g., empty metric names are skipped, attribute values are normalized, input metrics are not mutated). (`exporter/bmchelixexporter/internal/operationsmanagement/metrics_producer_test.go`, [[1]](diffhunk://#diff-e207bd2633bf323b83e6523cd0ba51f542e8f7008d33a654cd4b06d8d4943418L154-R263) [[2]](diffhunk://#diff-e207bd2633bf323b83e6523cd0ba51f542e8f7008d33a654cd4b06d8d4943418R442-R525)
* Test data is corrected to move `host.name` to resource attributes, reflecting real-world usage. (`exporter/bmchelixexporter/internal/operationsmanagement/metrics_producer_test.go`, [[1]](diffhunk://#diff-e207bd2633bf323b83e6523cd0ba51f542e8f7008d33a654cd4b06d8d4943418R125) [[2]](diffhunk://#diff-e207bd2633bf323b83e6523cd0ba51f542e8f7008d33a654cd4b06d8d4943418L135) [[3]](diffhunk://#diff-e207bd2633bf323b83e6523cd0ba51f542e8f7008d33a654cd4b06d8d4943418L144)

#### Documentation
* A detailed changelog entry is added, describing the enrichment and normalization changes, their rationale, and compatibility notes for BMC Helix Operations Management. (`.chloggen/bmchelix-enrich-metric-names.yaml`, [.chloggen/bmchelix-enrich-metric-names.yamlR1-R31](diffhunk://#diff-6b6abb47c7fdeb9ad4a91ab8626f4483d1c701721e45bff575a7ff645e0f9bd7R1-R31))

### CODE REVIEW

File: exporter/bmchelixexporter/internal/operationsmanagement/metrics_producer.go [hunk 1]
```diff
@@ -268,8 +298,9 @@ func (*MetricsProducer) updateEntityInformation(labels map[string]string, metric
 	// Convert metricAttrs from map[string]any to map[string]string for compatibility
 	stringMetricAttrs := make(map[string]string)
 	for k, v := range dpAttributes {
-		stringMetricAttrs[k] = fmt.Sprintf("%v", v)
-		labels[k] = fmt.Sprintf("%v", v)
+		normalizedValue := NormalizeLabelValue(fmt.Sprintf("%v", v))
```
Comment (MovieStoreGuy) [inline exporter/bmchelixexporter/internal/operationsmanagement/metrics_producer.go]: In this case, you can just call `fmt.Sprint`, no need for the formatting

**Review by MovieStoreGuy — COMMENTED** [review]
Look at the code, this should be configurable since you're now increasing the cardinality of data being sent.

File: exporter/bmchelixexporter/internal/operationsmanagement/metrics_producer.go (hunk 1 above)
Comment (NassimBtk) [reply exporter/bmchelixexporter/internal/operationsmanagement/metrics_producer.go]: Good catch! Thank you! I have fixed two other `fmt.Sprintf` BTW.

**Review by MovieStoreGuy — APPROVED** [review]
